src/environment.cpp

```diff
@@ -221,11 +222,12 @@ float Environment::getTimeOfDayF()
 
 void Environment::stepTimeOfDay(float dtime)
 {
-	// getTimeOfDaySpeed lock the value we need to prevent MT problems
-	float day_speed = getTimeOfDaySpeed();
+	MutexAutoLock(this->m_time_lock);
```

Comment:
to avoid too long locking please set braces on the portion of code which needs locking.